Update Docker solution

vsilverman opened the pull request:
Updating docker solution with more recent stuff

## After

![screencapture-localhost-4242-solutions-docker-2020-05-27-18_57_17-edit](https://user-images.githubusercontent.com/156685/83086608-78087400-a04c-11ea-98f5-ef7fd5caf346.png)

## Before

![screencapture-jenkins-io-solutions-docker-2020-05-27-19_04_43-edit](https://user-images.githubusercontent.com/156685/83086843-14cb1180-a04d-11ea-8095-726dc1afcbe1.png)

vsilverman commented:
This PR is linked to issue #3098

oleg-nenashev commented on content/solutions/docker.adoc:
```diff
@@ -19,7 +19,29 @@ ____
+== Getting Started
+
+* link:https://docs.docker.com/get-started/[Getting Started with Docker]
+* link:https://hub.docker.com/r/jenkins/jenkins[Jenkins CI/CD server in Docker]
```
```suggestion
* link:https://hub.docker.com/r/jenkins/jenkins[Official Jenkins master image for Docker]
```

oleg-nenashev commented on content/solutions/docker.adoc:
```diff
@@ -19,7 +19,29 @@ ____
+* link:https://hub.docker.com/r/jenkins/jenkins[Jenkins CI/CD server in Docker]
+* link:https://medium.com/@gustavo.guss/quick-tutorial-of-jenkins-b99d5f5889f2[Jenkins+Docker Tutorial]
+
+== Demos
```
Not sure we want to add that. I suspect all these demos are quite outdated.
CC @bitwiseman @dwnusbaum

dwnusbaum replied:
Yeah, we would probably prefer for users to use different approaches, but I don't know if there is any content that demonstrates the current best practices in this area.

vsilverman replied:
How about adding blogs related content, like the following:
`== Blogs

* link:https://www.jenkins.io/blog/2020/05/11/docker-windows-agents/#using-windows-docker-images/[Windows Docker agent]`

MarkEWaite commented:
I removed the "step 0" from the page and removed the dead video link that was pointed to an obsolete eBay video.  Replaced 4 videos with never video segments.

I did not fix the overlapping box that hides the "plugins" list because I was unable to find the code that generates it.  May yet find it on another attempt.

MarkEWaite commented on content/solutions/docker.adoc:
```diff
@@ -19,7 +19,29 @@ ____
+* link:https://hub.docker.com/r/jenkinsci/pipeline-as-code-github-demo/[GitHub organization folders]
+* link:https://hub.docker.com/r/jenkinsci/docker-workflow-demo/[build, test, and publish with Docker]
+
+== Presentations
+
+* link:https://www.youtube.com/watch?v=pMO26j2OUME[Run Jenkins in Docker Container]
+* link:https://www.youtube.com/watch?v=6tcoRIPBd8s[Jenkins - Build & Publish Docker Images]
+* link:https://www.youtube.com/watch?v=iN3he0eVUyw&t=31s[Jenkins on Docker: step by step for beginners]
+* link:https://www.youtube.com/watch?v=nMLQgXf8tZ0[DevOps project with Git, Jenkins and Docker on AWS]
+* link:https://www.youtube.com/watch?v=9TX5LOInPIQ[Deploy a war file on Docker container using Jenkins]
+* link:https://www.youtube.com/watch?v=z32yzy4TrKM[How to build Docker Image Using Jenkins Pipeline]
```
The top of the page already provides a location and thumbnails for videos.  Wouldn't it be better to include this in the video thumbnails above rather than a list of links here.

vsilverman commented:
Thanks for the guidance.  As discussed I:
- replaced outdated demos with more recent ones, outlining Jenkins+Docker best practicies
- removed videos from the bottom of page, leaving the ones at the top.
Latest After:
![image](https://user-images.githubusercontent.com/2159948/83203889-50beaf00-a0ff-11ea-8b68-a2d651424e70.png)
